lib/components/email-picker.js

```diff
@@ -0,0 +1,74 @@
+import {EmailAddressesTrigger} from "../logic/triggers";
+var React = require("react");
+
+var TriggerMixin = require("../logic/triggers").TriggerMixin([EmailAddressesTrigger]);
+
+module.exports = React.createClass({displayName: "EmailPicker",
+    mixins: [TriggerMixin],
+
+    getInitialState: function() {
+        return {
+            lastEmail: null
+        }
+    },
+
+    componentDidMount: function() {
+        this.setTrigger(new EmailAddressesTrigger());
+        this.props.controller.getEmailAddresses();
+    },
+
+    propTypes: {
+        controller: React.PropTypes.any.isRequired,
+        onSelect: React.PropTypes.func.isRequired
+    },
```

Comment:
Nit: I prefer to have `propTypes` and `getInitialState` at the top of the file (just beneath the `mixin`) so I can clearly see which variables we're playing around with.